src/prommis/uky/uky_flowsheet.py

```diff
@@ -1467,31 +1467,193 @@ def add_costing(flowsheet):
+        units.convert(
+            flowsheet.fs.roaster.flow_mol_comp_dust[0, "Ce"]
+            * 328.24
+            * units.g
+            / units.mol,
+            to_units=units.kg / units.hr,
+        )
+    )
+    dy_dust = value(
+        units.convert(
+            flowsheet.fs.roaster.flow_mol_comp_dust[0, "Dy"]
+            * 373
+            * units.g
+            / units.mol,
+            to_units=units.kg / units.hr,
+        )
+    )
+    gd_dust = value(
+        units.convert(
+            flowsheet.fs.roaster.flow_mol_comp_dust[0, "Gd"]
+            * 362.5
+            * units.g
+            / units.mol,
+            to_units=units.kg / units.hr,
+        )
+    )
+    la_dust = value(
+        units.convert(
+            flowsheet.fs.roaster.flow_mol_comp_dust[0, "La"]
+            * 325.82
+            * units.g
+            / units.mol,
+            to_units=units.kg / units.hr,
+        )
+    )
+    nd_dust = value(
+        units.convert(
+            flowsheet.fs.roaster.flow_mol_comp_dust[0, "Nd"]
+            * 336.48
+            * units.g
+            / units.mol,
+            to_units=units.kg / units.hr,
+        )
+    )
+    pr_dust = value(
+        units.convert(
+            flowsheet.fs.roaster.flow_mol_comp_dust[0, "Pr"]
+            * 329.82
+            * units.g
+            / units.mol,
+            to_units=units.kg / units.hr,
+        )
+    )
+    sc_dust = value(
+        units.convert(
+            flowsheet.fs.roaster.flow_mol_comp_dust[0, "Sc"]
+            * 137.912
+            * units.g
+            / units.mol,
+            to_units=units.kg / units.hr,
+        )
+    )
+    sm_dust = value(
+        units.convert(
+            flowsheet.fs.roaster.flow_mol_comp_dust[0, "Sm"]
+            * 348.72
+            * units.g
+            / units.mol,
+            to_units=units.kg / units.hr,
+        )
+    )
+    y_dust = value(
+        units.convert(
+            flowsheet.fs.roaster.flow_mol_comp_dust[0, "Y"]
+            * 225.812
+            * units.g
+            / units.mol,
+            to_units=units.kg / units.hr,
+        )
+    )
     mixed_product_output_rates = {
-        "Sc2O3": 0.00143 * units.kg / units.hr,
-        "Y2O3": 0.05418 * units.kg / units.hr,
-        "La2O3": 0.13770 * units.kg / units.hr,
-        "CeO2": 0.37383 * units.kg / units.hr,
-        "Pr6O11": 0.03941 * units.kg / units.hr,
-        "Nd2O3": 0.17289 * units.kg / units.hr,
-        "Sm2O3": 0.02358 * units.kg / units.hr,
-        "Eu2O3": 0.00199 * units.kg / units.hr,
-        "Gd2O3": 0.00000 * units.kg / units.hr,
-        "Tb4O7": 0.00801 * units.kg / units.hr,
-        "Dy2O3": 0.00000 * units.kg / units.hr,
-        "Ho2O3": 0.00000 * units.kg / units.hr,
-        "Er2O3": 0.00000 * units.kg / units.hr,
-        "Tm2O3": 0.00130 * units.kg / units.hr,
-        "Yb2O3": 0.00373 * units.kg / units.hr,
-        "Lu2O3": 0.00105 * units.kg / units.hr,
+        "CeO2": ce_dust * units.kg / units.hr,
```

Comment:
Looking at how this is used later, this appears to be wrong to me. Why are you passing static values to the costing, and not the actual variables?

Reply:
What's the issue with using static values instead of the variables? Shouldn't the two be equivalent at steady state?

Reply:
No - the static value is only valid at the time you access it. Most importantly, it means if you try to do an optimization these values will not change; i.e. you can change the actual flowsheet performance but the costs will not change to reflect this.